org.eclipse.winery.lsp/src/main/java/org/eclipse/winery/lsp/Server/ServerCore/TOSCAFileValidator.java

```diff
@@ -0,0 +1,141 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache Software License 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+
+package org.eclipse.winery.lsp.Server.ServerCore;
+
+import org.yaml.snakeyaml.error.Mark;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Set;
+
+public class TOSCAFileValidator {
+    public ArrayList<TOSCAFileDiagnostics> diagnostics = new ArrayList<>();
+
+    public void validateRequiredKeys(Map<String, Object> yamlMap, Path path) {
+        if (!yamlMap.containsKey("tosca_definitions_version")) {
+            handleDiagnosticsError("Missing required key: tosca_definitions_version", path);
+        }
+    }
+
+    public void validateRequiredKeys(Map<String, Object> yamlMap, String content) {
+        if (!yamlMap.containsKey("tosca_definitions_version")) {
+            handleDiagnosticsError("Missing required key: tosca_definitions_version", content);
+        }
+    }
+
+    public void validateKeywords(Map<String, Object> yamlMap, Map<String, Mark> positions, String YamlContent) {
+        Set<String> validKeywords = Set.of(
+            "tosca_definitions_version", "description", "metadata", "dsl_definitions",
+            "artifact_types", "data_types", "capability_types", "interface_types",
+            "relationship_types", "node_types", "group_types", "policy_types",
+            "repositories", "functions", "profile", "imports", "service_template"
```

Comment:
Don't spend much time in validation. 

Auto complete and Ctrl+click on a thing (go-to definition) is much more important!